ompi/communicator/comm.c

```diff
@@ -2690,6 +2689,49 @@ static int ompi_comm_copy_topo(ompi_communicator_t *oldcomm,
     return OMPI_SUCCESS;
 }
 
+int ompi_comm_set_disjointness(ompi_communicator_t *newcomm, ompi_communicator_t *oldcomm)
+{
+    int local_peers = 0, rc = OMPI_ERROR;
+
+    if (OMPI_COMM_IS_DISJOINT_SET(newcomm)) {
+        rc = OMPI_SUCCESS;
+        goto out;
+    }
+
+    if (NULL != oldcomm && OMPI_COMM_IS_DISJOINT(oldcomm)) {
+        /**
+         * A communicator splitted from a disjoint
+         * communicator(1 process per node) is also disjoint
+         */
+        newcomm->c_flags |= (OMPI_COMM_DISJOINT_SET | OMPI_COMM_DISJOINT);
+        rc = OMPI_SUCCESS;
+        goto out;
+    }
+
+    if (!newcomm->c_coll) {
+        rc = OMPI_ERR_NOT_AVAILABLE;
+        goto out;
+    }
+
+    local_peers = ompi_group_count_local_peers(newcomm->c_local_group);
+    rc = newcomm->c_coll->coll_allreduce(MPI_IN_PLACE, &local_peers, 1, MPI_INT, MPI_MAX, newcomm,
+                                         newcomm->c_coll->coll_allreduce_module);
+    if (OPAL_UNLIKELY(OMPI_SUCCESS != rc)) {
+        goto out;
+    }
+
+    if (1 == local_peers) {
+        newcomm->c_flags |= OMPI_COMM_DISJOINT;
+    } else {
+        newcomm->c_flags &= ~OMPI_COMM_DISJOINT;
```

Comment:
I don't think this is necessary, the flags are build up and I cannot imagine a scenario in which the DISJOINT flag was set but need to be removed (except process migration but that's a different topic).

Reply:
I see. I might have been too defensive in this case - I will remove this branch.